### DISCUSSION TITLE
Esp32 with modbus Rs485

### DISCUSSION THREAD

**Discussion by aymandab1** [espressif/arduino-esp32#7678] [Q&A]
Hello all, I need help. I have a Modbus RS485 sensor and I want to read its value from an ESP32. I have searched a lot for codes and libraries, such as SoftwareSerial and ModbusMaster. If anyone can help me and provide a code for reading data from the Modbus sensor using an ESP32, I would greatly appreciate it.

Baudrate: 9600 
Function: read holding registers x4
9 bits 
Id : 1

**Comment by aymandab1** [comment]
I am using MAX485, any help?

**Comment by pedrominatel** [comment]
Hi,

I'm not sure about Modbus libs for Arduino, however, you can use ESP-IDF code on Arduino.
Please take a look at the ESP-IDF examples here: [Modbus Serial](https://github.com/espressif/esp-idf/tree/master/examples/protocols/modbus/serial).

**Comment by bertmelis** [comment]
https://github.com/eModbus/eModbus

Mind that there might be some quirks when using a simple max485.
